ruma-identifiers/src/mxc_uri.rs

```diff
@@ -98,22 +97,52 @@ impl serde::Serialize for MxcUri {
 
 #[cfg(test)]
 mod tests {
-    use std::convert::TryFrom;
+    use matches::assert_matches;
 
     use super::MxcUri;
 
     #[test]
     fn parse_mxc_uri() {
-        assert!(<MxcUri>::try_from("mxc://127.0.0.1/asd32asdfasdsd").is_ok());
+        let mxc = <MxcUri>::from("mxc://127.0.0.1/asd32asdfasdsd");
+
+        assert!(mxc.is_valid());
+        assert_matches!(mxc.media_id(), Some("asd32asdfasdsd"));
+        assert_matches!(mxc.server_name(), Some(name) if name.as_str() == "127.0.0.1")
     }
 
     #[test]
     fn parse_mxc_uri_without_media_id() {
-        assert!(!<MxcUri>::try_from("mxc://127.0.0.1").is_ok());
+        let mxc = <MxcUri>::from("mxc://127.0.0.1");
+
+        assert!(!mxc.is_valid());
+        assert_matches!(mxc.media_id(), None);
+        assert_matches!(mxc.server_name(), None)
     }
 
     #[test]
     fn parse_mxc_uri_without_protocol() {
-        assert!(!<MxcUri>::try_from("127.0.0.1/asd32asdfasdsd").is_ok());
+        assert!(!<MxcUri>::from("127.0.0.1/asd32asdfasdsd").is_valid());
+    }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn serialize_mxc_uri() {
+        assert_eq!(
+            serde_json::to_string(&MxcUri::from("mxc://server/1234id"))
+                .expect("Failed to convert MxcUri to JSON."),
+            r#""mxc://server/1234id""#
+        );
+    }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn deserialize_mxc_uri() {
+        let mxc = serde_json::from_str::<MxcUri>(r#""mxc://server/1234id""#)
+            .expect("Failed to convert JSON to MxcUri");
+
+        assert_eq!(mxc.as_str(), "mxc://server/1234id");
+        assert!(mxc.is_valid());
+        assert_matches!(mxc.media_id(), Some("1234id"));
+        assert_matches!(mxc.server_name(), Some(name) if name.as_str() == "server")
```

Comment:
I think here `assert_eq!` would be easier to read, I only use `assert_matches!` when there is no `PartialEq` implementation for `assert_eq!` to use.